app/questionnaire/navigator.py

```diff
@@ -90,11 +87,9 @@ def build_path(self, blocks, this_location):
 
             # No routing rules, so if this isn't the last block, step forward a block
             elif block_index < len(blocks) - 1:
-                this_location = {
-                    "block_id": blocks[block_index + 1]['block']['id'],
-                    "group_id": blocks[block_index + 1]['group_id'],
-                    "group_instance": blocks[block_index + 1]['group_instance'],
-                }
+                this_location = Location(blocks[block_index + 1]['group_id'],
```

Comment:
Not sure what is being asked here, either solution seems to be more complex than what is in place already.